criu/cgroup.c

```diff
@@ -1652,20 +1675,28 @@ static int prepare_cgroup_sfd(CgroupEntry *ce)
 	pr_info("Preparing cgroups yard (cgroups restore mode %#x)\n",
 		opts.manage_cgroups);
 
-	off = sprintf(paux, ".criu.cgyard.XXXXXX");
-	if (mkdtemp(paux) == NULL) {
-		pr_perror("Can't make temp cgyard dir");
-		return -1;
-	}
+	if (opts.cgroup_yard) {
+		off = sprintf(paux, "%s", opts.cgroup_yard);
 
-	cg_yard = xstrdup(paux);
-	if (!cg_yard) {
-		rmdir(paux);
-		return -1;
-	}
+		cg_yard = xstrdup(paux);
+		if (!cg_yard)
+			return -1;
+	} else {
+		off = sprintf(paux, ".criu.cgyard.XXXXXX");
+		if (mkdtemp(paux) == NULL) {
+			pr_perror("Can't make temp cgyard dir");
+			return -1;
+		}
 
-	if (make_yard(cg_yard))
-		goto err;
+		cg_yard = xstrdup(paux);
+		if (!cg_yard) {
+			rmdir(paux);
+			return -1;
+		}
+
+		if (make_yard(cg_yard))
+			goto err;
+	}
 
```

Comment:
a few lines below we do:
```C
/* Create controller if not yet present */
if (access(paux, F_OK)) {
	pr_debug("\tMaking controller dir %s (%s)\n", paux, opt);
	if (mkdir(paux, 0700)) {
		pr_perror("\tCan't make controller dir %s", paux);
		return -1;
	}
	if (mount("none", paux, "cgroup", 0, opt) < 0) {
		pr_perror("\tCan't mount controller dir %s", paux);
		return -1;
	}
}
```
should we skip this step when `opts.cgroup_yard` is used?

Reply:
If external yard is passed then `access` will return 0 and we don't run the code inside this `if`. I could add another `if` here checking if external yard was passed but I don't really see the reason since nothing would change.